daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -55,7 +60,14 @@ object Speedy {
       var globalDiscriminators: Set[crypto.Hash],
   ) {
 
-    def kontPop(): Kont = kont.remove(kont.size - 1)
+    def pushKont(k: Kont): Unit = {
+      kontStack.add(k); ()
```

Comment:
:+1:

Reply:
Alos you could add `@inline` anotation to this kind of small utility methods. does not hurt.

Reply:
It might be interesting to make an experimental PR where we go crazy with @inline and see if we can actually measure any different it makes.

Reply:
I would prefer if we do refactorings like this one not in PRs where we actually try to change something. It makes reviewing so much harder because I need to find the real changes among a lot of noise. Same for renaming `Machine.kont` to `Machine.kontStack`.

Reply:
@nickchapman-da Let's please move these refactorings into a separate PR that goes to master first. This PR is controversial and I would like to focus on what's really going on.

Reply:
I don't understand why you think this PR is controversial. Can you explain your concerns?

By refactorings, I see:
- rename machine member: `kont` -> `kontStack`
- rename method: `kontPop` -> `popKont`
- abstracted new method: `kont.add` -> `pushCont`
- addition of 4 `@inline` annotations

Is there anything else you consider a refactoring?

I'll extract these into a pre-PR as you request.

Reply:
Yes, that's all the refactoring I've noticed.

The PR is controversial because we're changing the architecture of the interpreter by introducing recursion into the main loop. Even if it is bounded, it is still recursion. I would prefer if we find a way to get the gains without using recursion.

Reply:
But @remyhaemmerle-da argued that bounded stack usage is preferable to heap usage. (unless I misunderstood him)

Reply:
I generally agree with @remyhaemmerle-da's sentiment. However, the conclusion I draw is not "use recursion and more stack" but rather "use less heap and try to avoid recursion".